wallet_test.go

```diff
@@ -0,0 +1,121 @@
+// Package implements a wallet, that holds bitcoin and operations: to deposit and withdraw money and get the wallet's balance
+
+package wallet
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestWallet_GetBalance(t *testing.T) {
+	tests := []struct {
+		name string
+		w    *Wallet
+		want Bitcoin
+	}{
+		{
+			name: "zero balance",
+			w:    &Wallet{balance: 0},
+			want: 0,
+		},
+		{
+			name: "positive balance",
+			w:    &Wallet{balance: 1},
+			want: 1,
+		},
+		{
+			name: "negative balance",
+			w:    &Wallet{balance: -1},
+			want: -1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.w.GetBalance(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Wallet.GetBalance() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestWallet_Withdraw(t *testing.T) {
+	type args struct {
+		b Bitcoin
+	}
+	tests := []struct {
+		name    string
+		w       *Wallet
+		args    args
```

Comment:
unnecessary complexity. You have 1 arg, just do `withdrawAmmount Bitcoin`, smth like that

Reply:
Done

Reply:
check the whole code

Reply:
Done